### ISSUE TITLE
3.2a questions and comments

### ISSUE THREAD

**Issue by nicm** [tmux/tmux#2737]
### Ask or discuss any quick questions here about 3.2a please.

Bug reports please open a new issue with full information, see [CONTRIBUTING](https://github.com/tmux/tmux/blob/master/.github/CONTRIBUTING.md).

**[The 3.2a CHANGES file is here](https://raw.githubusercontent.com/tmux/tmux/3.2a/CHANGES).**

The previous issue for 3.2 is [here](https://github.com/tmux/tmux/issues/2645).

As always, thanks to the many people who contributed to both this and to previous releases.

---

Changes from 3.2 to 3.2a are:

* Add an "always" value for the "extended-keys" option; if set then tmux will forward extended keys to applications even if they do not request them.

* Add a "mouse" terminal feature so tmux can enable the mouse on terminals where it is known to be supported even if terminfo(5) says otherwise.

* Do not expand the filename given to -f so it can contain colons.

* Fixes for problems with extended keys and modifiers, scroll region, source-file, crosscompiling, format modifiers and other minor issues.

---

There are a **lot** of changes in the 3.2 release, these are the most notable, for more see the CHANGES file:

- How data is sent to control mode clients has been completely revamped to both be more fair with multiple panes and to prevent huge amounts of data being backed up; this feature is used by default when tmux is attached with the latest iTerm2 nightly builds.

- Configuration file parsing has changed slightly: the contents of the new {} syntax introduced in 3.1 must now be valid tmux command syntax; and to allow formats to be annotated, strings given with quotes may now contain newlines (leading spaces and comments are stripped).

- A new customize mode available with `C-b C` (`C-b S-c`) which allows options and key bindings to be browsed and changed interactively.

- Support for extended keys offered by some terminals (xterm, mintty, iTerm2), see https://github.com/tmux/tmux/wiki/Modifier-Keys#extended-keys.

- A `pane-border-lines` option to change the characters used to draw the pane border separators.

- How UTF-8 data is stored has been rewritten to reduce memory use for characters in the BMP.

- The message log (`C-b ~`) has been changed to be per server instead of per client and to have some useful content.

- Client flags have been extended (`refresh-client -F` becomes `-f` and also added for `attach-session` and `new-session`), and a `read-only` flag added (same as `-r` to `attach-session`).

- A new `active-pane` client flag that if given allows a client to have its own active pane for each window rather than being tied to the server's active pane.

- Improved `command-prompt` completion including showing a menu of completions.

- All style options can now be formats, for example the default `pane-active-border-style` now changes colour depending on `pane_in_mode` and `synchronize-panes`.

- Search match count has come back in copy mode, as well as performance improvements and additional styles for marking of search terms.

- Try `$XDG_CONFIG_HOME/tmux/tmux.conf` as well as `~/.config/tmux/tmux.conf`.

- Window and pane hooks such as `window-layout-changed` and `pane-exited` are now window or pane options instead of session options.

- `copy-pipe` and friends can now be given with no arguments in which case they get the command from a new `copy-command` option. The default copy commands all now use `copy-pipe`. This makes it much easier to configure `copy-pipe` for all copy key bindings.

- Terminal detection and feature code has been rewritten. tmux now collects sets of related terminfo(5) capabilties together with sensible default values into named "terminal features" which makes them easier to configure via a new `terminal-features` option or `-T` flag to tmux itself.  tmux can detect and will automatically configure a number of common, modern terminal emulators.

- Support for per-client overlay popups with a new `display-popup` command. A popup is also used for an `e` key binding in buffer mode to edit a buffer using the editor given in the new `editor` option (filled in from the `EDITOR` or `VISUAL` environment variables by default).

- `DoubleClick` and `TripleClick` mouse key bindings have changed so that only one is fired. Behaviour similar to the old double-click key binding is now available as `SecondClick`. There are now default double- and triple-click bindings to copy a word or line, and a default middle-click binding to paste the top buffer.

- When a server socket is given with `-S`, it is now created with umask 177 rather than 117.

---

**Note the following known potentially-incompatible changes:**

- The {} syntax added in tmux 3.1 has changed so that the contents must follow the syntax of tmux commands.

- The `DoubleClick` key binding is not fired on the way to `TripleClick`; only one is fired. The new `SecondClick` key binding behaves similarly to the old `DoubleClick`.

- `status-fg` and `status-bg` are now applied before `status-style` rather than instead of; this means that if either is set to `default` the `status-style` colour will be used. To fix, replace `status-bg default` with `status-style bg=default`.

**Comment by Stikus** [comment]
Hello, I'm not sure should I create a separate issue or ask here coz my question is not related to 3.2a exactly - is there any method to create multiline status bar with panes separated between lines?
| ssh1 | ssh2 | ssh3 |
| ----- | ----- | ----- |
| ssh4 | ssh5 | ssh6 |

Like this. And new tabs appear on selected row.

**Comment by nicm** [comment]
You can probably show first N panes on status[0] and second N on status[1] and so on but it will be difficult to make it dynamic (that is, you will need to pick N up front). You can use the P format operator to loop over panes.

**Comment by Stikus** [comment]
> You can probably show first N panes on status[0] and second N on status[1] and so on but it will be difficult to make it dynamic (that is, you will need to pick N up front). You can use the P format operator to loop over panes.

I'm fine with static N (i.e. 10 panes) but I'm struggling with finding `status-format` examples.
Where can I found any good info about this?

I'm good if first status line will contain 10 panes and second line will contain all default info and all panes besides first ten.
I know that I need `set -g status 2` for 2 lines and separate format for each, like:
`set -Fg 'status-format[1]' '#{status-format[0]}'` but I don't know what to write in this format.

**Comment by nicm** [comment]
https://github.com/tmux/tmux/wiki/Formats

Also the stuff about formats and styles in the man page.

For example you can show only the first 3 panes with something like:
~~~~
#{P:#{?#{e|<|:#{pane_index},3},#{pane_index} #{pane_title} ,}}
~~~~

**Comment by kylo252** [comment]
Not sure if this warrants a new issue or if it was discussed before but nothing really came up during my search.

Is it possible for the popup to behave more "asynchronously"? Here's a simple test, hopefully it's clear enough.

![tmux-async](https://user-images.githubusercontent.com/59826753/121703650-58b37c00-cad3-11eb-8e71-ef6d41102044.gif)

**Comment by nicm** [comment]
No, popups take over the whole client until they are done.

**Comment by kylo252** [comment]
> No, popups take over the whole client until they are done.

Is it a limitation or more of a design decision? I'm sure a few others would find such a feature useful as well, if it's feasible of course.

**Comment by David-Else** [comment]
I am a bit confused by the new syntax, this works:
```
bind g run -C { popup -EE -w 100% -h 100% "lazygit -p #{pane_current_path}" }                  
bind t popup -E -w 60% -h 100% "[[ -e TODO.md ]] && nvim TODO.md || nvim ~/Documents/TODO.md"  
```
but am I using old syntax? Is the above the best way to write this with the latest version? Could this be done better with `display-popup`?

**Comment by arekm** [comment]
Not related to only this release but also many before.

Any ideas why tmux when started on 256 capable terminal (like TERM=xterm-256color) doesn't automatically use 256 colors? Example

````
~ % echo $TERM
xterm-256color
~ % tmux
% echo $TERM
screen
````

I would expect screen-256color to be used automatically.

Currently tmux requires manual setting in config or starting with tmux -2 (but "-2" means "force" while it shouldn't be forced it it knows already that current terminal supports 256 colors due to knowing that TERM == xterm-256color)

Having to configure it each time, over and over, is really ... problematic.

**Comment by gotbletu** [comment]
**Q1: is it possible to use tmux paste while in tmux display-popup?**
e.g bind-key -n C-M-Space display-popup -w 80% -E "ls | fzf"
can't use tmux paste-buffer

**Q2: can tmux display-popup use environment variable of the last pane/window?**
e.g bind-key -n C-M-Space display-popup -w 80% -E "tmux save-buffer /tmp/buffer && urlview /tmp/tmux-buffer"

**Comment by vbauerster** [comment]
@nicm Can you please add `-e` switch for `display-popup` command?

```
-e takes the form `VARIABLE=value' and sets an environment variable for the newly created window; it may be specified multiple times.
```

Use case is that popup may be launched not from interactive prompt by means of terminal editor for example. In that case some important env vars cannot be passed to display-popup unlike split-window command.

**Comment by lijunzh** [comment]
I upgrade from tmux 3.2 to 3.2a this morning, but it fails to start with the following message on two of my Macs (11.4 and 10.15.7) using iTerm2. The same error can be reproduced in Terminal.app, Kitty, Alacritty as well.

```
[exited]
^[[>0;95;0c^[P>|iTerm2 3.4.8^[\%
```

**Comment by nicm** [comment]
@arekm not everyone has `screen-256color` and it is less confusing not to change it depending where you attached from.

@gotbletu no, they are not panes and take over the whole client, you can only paste by doing eg `tmux pasteb`.

@lijunzh if `pkill tmux` doesn't fix it, your `default-shell` or `default-command` is probably broken.

**Comment by nicm** [comment]
@gotbletu I would like to support some key bindings in when a popup is open to allow pasting and a few other tasks, or maybe a right-click menu. It would not be too hard to implement this.

**Comment by nicm** [comment]
@vbauerster added to todo list.

**Comment by lijunzh** [comment]
> `pkill tmux`

@nicm , this fix my issue. Thank you very much.

**Comment by camgraff** [comment]
Is it possible to get `display-popup` to automatically exit if the running process is suspended?

**Comment by nicm** [comment]
You want it to kill the program if it is suspended?

**Comment by kylo252** [comment]
On a related note, how can we go about doing something like this for e,g: an interactive shell: 

```tmux
# bind -n M-q display-popup -C
bind M-q display-popup -C
```

Or is the only solution to use a `readline` bind?

**Comment by camgraff** [comment]
> You want it to kill the program if it is suspended?

No, I just want to close the popup if the shell command passed to `display-popup` is suspended. The same behavior that the `-E` flag has now, but for suspension as well. I don't want the program to be killed, because I want to resume the suspended process later.

**Comment by nicm** [comment]
How would you resume it if the popup is closed? Closing the popup will kill it.

**Comment by camgraff** [comment]
I see. Yeah I guess this wont quite work then :upside_down_face: haha thanks

**Comment by munenari** [comment]
> `pkill tmux`
this helps me, thanks a lot!

**Comment by David-Else** [comment]
Any idea when https://galaxy4.net/repo/ will update to 3.2a?

**Comment by systemmonkey42** [comment]
Hi,

I'm having unexpected behaviour with 'set-option'

Its probably normal behaviour, but I can't see why its doing what it is.

My specific example is as follows;

- I use a key binding to "enable" or "disable" input to a pane.
- I defined a command-alias which includes a macro to select the next non-disabled pane.
- I defined a binding to trigger the command alias.

eg.
```
# Toggle "disabled" state via prefix `
bind-key -T prefix  '``''  if-shell -F "#{pane_input_off}" "select-pane -e" "select-pane -d"

# Select next "active" pane (active in this context means not disabled by 'select-pane -d')
set-option -sa command-alias "next-active-pane=if-shell -F #{P:#{?#{pane_input_off},,1}} { select-pane -t '{next}'; if-shell -F \"#{pane_input_off}\" { next-active-pane } }"

# \u0173 is what my keyboard sends for the key I use for "next-pane"
bind-key -T root   \u0173  next-active-pane
```

This does not work as expected. 

`tmux show-options -s command-alias` shows 
```
command-alias[6] "next-active-pane=if-shell -F \"#{P:#{?#{pane_input_off}"
command-alias[7] "1}}\" { select-pane -t '{next}'; if-shell -F \"#{pane_input_off}\" { next-active-pane } }"
```
Which is not what I expected.

I found if I select an alias Index, as in
```
set-option -s command-alias[100] "next-active-pane=if-shell -F #{P:#{?#{pane_input_off},,1}} { select-pane -t '{next}'; if-shell -F \"#{pane_input_off}\" { next-active-pane } }"
```
The result is more as I expected.

**Comment by nicm** [comment]
If you set an array option without specifying an index, it assumes you are giving a list of elements for the array. For `command-alias`, elements are separated by `,`.

**Comment by systemmonkey42** [comment]
Ah, so its not the "-a" option.

Is there any way to escape the "," in this context?   I've tried `#`s and blackslashes but nothing works.

However, I've also realised specifying an index is also safer when reloading tmux.conf, otherwise you end up with multiple definitions for the same command alias, and only the oldest is used.

**Comment by nicm** [comment]
No escaping, you need to specify an index.

**Comment by rcywongaa** [comment]
Will tmux support specifying alpha values for background colors (for background transparency)?  So far, the only method of supporting background transparency is to use the default color provided by the terminal which means the color cannot be further customized .

**Comment by nicm** [comment]
No, there is no way this can be emulated per pane. If you want to change it for the entire terminal, you can use the passthrough sequence.

**Comment by christoomey** [comment]
Hi there 👋 I manage a plugin for Vim & tmux navigation and I've been getting some issues coming in, seemingly related to a change in the ~3.2 range.

The issue of interested is: https://github.com/christoomey/vim-tmux-navigator/issues/299#issuecomment-870130970

End users seem to be experiencing timeouts of ~30 seconds when using variations of `if-shell` (that's my best read on what's happening, but I'm far from certain). Wanted to share in case anyone could shed some light, or possibly to highlight if there might be an issue with our usage. Any pointers would be greatly appreciated.

**Comment by nicm** [comment]
Open an issue with logs please.

**Comment by systemmonkey42** [comment]
Hi,

I have a questions regarding popups in the latest build (master, build a few minutes ago for testing)

I have two clients connected, and I want a popup to appear on the OTHER client.

The `-t` target parameter isn't working as expected.
```
$ tmux list-clients
/dev/pts/0: 0 [136x70 xterm-256color] (attached,focused,UTF-8)
/dev/pts/2: 1 [136x70 xterm-256color] (attached,UTF-8)
```

All the commands below open the popup on the current client without error.  If I specify an invalid target I get an error. 
 
```
tmux popup 'top'
tmux popup -t /dev/pts/0 'top'
tmux popup -t /dev/pts/2 'top'
tmux popup -t pts/0 'top'
tmux popup -t pts/2 'top'
```

Any ideas?
Thanks

**Comment by nicm** [comment]
~~~~
     display-popup [-CE] **[-c target-client]** [-d start-directory] [-h height]
                            ^^

**Comment by systemmonkey42** [comment]
> ```
>      display-popup [-CE] **[-c target-client]** [-d start-directory] [-h height]
>                             ^^
> ```
Yeah, this is embarrassing.  Thanks for pointing out what should have been obvious.

**Comment by ClashTheBunny** [comment]
I have been using display-popup for a while as a little bit of a quaketerm.  Because I want it to be persistent, I actually start a tmux in the display popup.  Because I want to use this other places, I want to name the tmux in the display-popup something associated with my current session name.  This used to work:
```
bind '`' display-popup -d '#{pane_current_path}' -xC -yC -w'90%' -h'90%' -K -E -R "tmux attach -t #S-popup || tmux new -s #S-popup"
```
I would get a tmux named `caesar-popup` on my `caesar` tmux popup and a tmux named `tsar-popup` on my `tsar` tmux session.  Since I use a shared computer most of the time, I can't just call it `popup`.  Since I sometimes have two tmux sessions open, I can't do something static like my username.

When I set it up with the new syntax, I don't thing the format is being interpreted anymore:
```
bind '`' display-popup -d '#{pane_current_path}' -xC -yC -w'90%' -h'90%' "tmux attach -t #S-popup || tmux new -s #S-popup"
```

I've tried a few different combinations, like adding `--` before the command.  I can't figure out if it's something I've done, or if it's a subtle bug.  I'm on debian's current experimental package:
```
~/code ❯❯❯ dpkg -l | grep -i tmux
ii  tmux                                             3.2a-1                                    amd64        terminal multiplexer
```

**Comment by JustSage** [comment]
Hello,

The configuration file is not loaded from the $XDG_CONFIG_HOME. 
My TMUX_CONF is located in a .zshenv file, could this be the reason? 
How should one debug this :)?

**Comment by blueyed** [comment]
Should the default binding for `?` use `list-keys -Na` maybe to display notes, but still include/list other mappings?
I was confused for custom bindings not being listed there anymore.

Relevant changelog entries:

* Add -a to list-keys to also list keys without notes with -N.

* Add support for adding a note to a key binding (with bind-key -N) and use
  this to add descriptions to the default key bindings. A new -N flag to
  list-keys shows key bindings with notes. Change the default ? binding to use
  this to show a readable summary of keys. Also extend command-prompt to return
  the name of the key pressed and add a default binding (/) to show the note
  for the next key pressed.

**Comment by nicm** [comment]
I don't know, I think it is equally confusing to include them because then don't you get a bunch of mouse key bindings by default...

**Comment by systemmonkey42** [comment]
> My TMUX_CONF is located in a .zshenv file, could this be the reason?

If you are suggesting you set "$TMUX_CONF" to point to a configuration file, that is not how it currently works.

@nicm , Would it be reasonable to make $TMUX_CONF runtime feature rather than just a build option? 

I tried adding `$$TMUX_CONF` to Makefile.am (-DTMUX_CONF=$$TMUX_CONF:.....) but when I `export TMUX_CONF=/tmp/test`, the file /tmp/test gets loaded first, followed by all the standard config files.

I would like to suggest accepting "$TMUX_CONF" as an alias for `-f <configfile>`

Thanks

**Comment by nicm** [comment]
I think specifying config file with an environment variable is unusual and unexpected and I don't see what use it really has.

**Comment by bfrg** [comment]
> - Try $XDG_CONFIG_HOME/tmux/tmux.conf as well as ~/.config/tmux/tmux.conf.

This is not mentioned in `tmux(1)`. I think it makes sense to add it under the `FILES` section, and maybe under the `-f` option as well.

**Comment by jalal** [comment]
When I run 
```
$tmux list-keys
```
I get the expected listing of assigned keys, including copy-mode keys. However, when I do `c-a ?` I only get a list of normal mode keys.
Is this an expected change or is something amiss?

**Comment by bfrg** [comment]
@jalal <kbd>c-a</kbd><kbd>?</kbd> is like running `tmux list-keys -N`. This was changed in 3.1, see the [CHANGES](https://github.com/tmux/tmux/blob/63f9f3113c731b7a7ced2f10b047722647d7b544/CHANGES#L441-L444) file. It looks like it's not mentioned in the man page.

**Comment by b0o** [comment]
I'm trying to display a message on all clients. When I try to specify a client with `-c`, the `display-message` command fails and returns the usage message. Example:

```
$ tmux list-clients -F "#{client_name}"
/dev/pts/1
/dev/pts/77
/dev/pts/80
/dev/pts/66
/dev/pts/49
/dev/pts/52
/dev/pts/84
/dev/pts/35
/dev/pts/4
/dev/pts/44
/dev/pts/82
/dev/pts/5
/dev/pts/102
/dev/pts/100
/dev/pts/113
/dev/pts/119
/dev/pts/63

$ tmux display-message -c /dev/pts/63 test
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message] 

$ tmux list-clients -F "#{client_name}" | xargs -i tmux display-message -c '{}' "test" 
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
usage: display-message [-aINpv] [-c target-client] [-d delay] [-F format] [-t target-pane] [message]
```

Am I doing something wrong or is this a bug?

**Comment by nicm** [comment]
This is a bug, try making this change near the top of cmd-display-message.c:
~~~~
-       .args = { "acd:INpt:F:v", 0, 1 },
+       .args = { "ac:d:INpt:F:v", 0, 1 },

**Comment by b0o** [comment]
Yes that fixes the issue. Glad to know it wasn't me, I was going a bit crazy trying to figure out what I was doing wrong.

**Comment by nicm** [comment]
Great, fixed now, thanks!

**Comment by systemmonkey42** [comment]
Hey,   All my `command-aliase`s stopped worked when I build the latest version and I'm not sure how to fix them.

I have (among others)

```
set-option -s command-alias[102] "csplit-window=split-window -c \"#{pane_current_path}\""

... later in the file

bind-key -T prefix       |                               csplit-window -h
```

With a previous build I woud get
```
$ tmux-3.2 list-key -P prefix '|'
bind-key -T prefix | split-window -h -c "#{pane_current_path}"
```
This works correctly to vertically split a window and open a new pane in the same directory.

As of commit 84955e3d

Using the `list-keys` command I see:
```
$ tmux list-key -P prefix '|'
bind-key -T prefix | split-window -c -h "#{pane_current_path}"
```
This does not work, because the "-h" is now inserted between the `-c` and its argument.

Any suggestions?

Thanks

**Comment by nicm** [comment]
Open a new issue please.

**Comment by k8xst08htbn78gcd** [comment]
Is there some estimated date for follow-up release, 3.3?
Looking forward to its being available via "brew update".
( for fix for a "3.2a" issue https://github.com/tmux/tmux/issues/2749 )

**Comment by systemmonkey42** [comment]
> Thanks. The plan is to have a status pane on the right side. I guess I would have to create an environment variable to only have it trigger on the first split, although that would make it not work on new sessions.... Thanks for the solution, I'll try to figure out these new problems.

You answered your own question...
```
set-hook -g after-new-session "split-window -h htop"
```
Here, every new session gets a full size vertical split running "htop" (on the first window).   If you want one on every window, thats a bit messier.

**Comment by jpie726** [comment]
> > Thanks. The plan is to have a status pane on the right side. I guess I would have to create an environment variable to only have it trigger on the first split, although that would make it not work on new sessions.... Thanks for the solution, I'll try to figure out these new problems.
> 
> You answered your own question...
> 
> ```
> set-hook -g after-new-session "split-window -h htop"
> ```
> 
> Here, every new session gets a full size vertical split running "htop" (on the first window). If you want one on every window, thats a bit messier.

Thank you much, that's just what I've been looking for. I guess RTFM does work :sweat_smile:

**Comment by ElijahLynn** [comment]
Rubber ducking here: An update broke all my keybindings in .tmux.conf. I reading through everything now, and so far I see that not even `C-b C` which I had changed to `C-b \` doesn't work either. Maybe something to do with escapes. Will post updates to this comment.

**Comment by nicm** [comment]
Use `terminal` instead of `default` to use termin default colour instead of inheriting.

**Comment by listx** [comment]
> Use `terminal` instead of `default` to use terminal default colour instead of inheriting.

Thank you, this works exactly as expected! Sorry for the noise.

**Comment by tigerinus** [comment]
Is there a way to bind mouse right-click to `Ctrl`-`B`+`]` (paste)?

**Comment by daiaji** [comment]
Does enabling the XDG specification require enabling FLAG during compilation?

**Comment by ThomasAdam** [comment]
> Does enabling the XDG specification require enabling FLAG during compilation?

@daiaji -- no  You just need to have `XDG_CONFIG_HOME` defined, and put `tmux/tmux.conf` relative to that.  It's all in `man tmux`.

**Comment by fawaf** [comment]
upgraded to v3.2a and for some reason piping of tmux commands stopped working.

works fine:
```
% tmux list-sessions
*aws: 1 windows (created Fri Feb  4 18:16:28 2022)
%
```

just hangs and does not work:
```
% tmux list-sessions | grep foo
```

**Comment by fawaf** [comment]
oh, nvm. a `tmux kill-server` fixed the issue. odd.....

**Comment by igorclark** [comment]
Hi there, I have an intermittent segfault happening in 3.2a on macOS 10.15.7 that I don't know how to reproduce. It seems like the only way to make it happen is be right in the middle of working on something important and then accidentally hit some unknown key combination. That totally kills it, and it takes all the child processes and panes and editors and shells and everything with it 🤷🏻‍♂️

I have a crash log, but interpreting it is a bit above my pay grade. I can see there's `crashed on child side of fork pre-exec` which might be important, also that the macOS console says that `alacritty` is `responsible`, but it doesn't seem to be an alacritty crash, 'cos when that happens the tmux server stays alive and I can just pick it up with `attach`.

Should I paste the log in here, start a new issue, something else?

Thanks 👍

**Comment by nicm** [comment]
Does it still happen if you build tmux from master? There is at least one
know crash in 3.2.

**Comment by igorclark** [comment]
Thanks @nicm, noooo, I've only ever installed from `brew`. Maybe they're a bit aggressive in bundling up `3.2a`? Ah, or, do you mean master/3.3 might fix it?

**Comment by nicm** [comment]
Yes I mean master might fix it.

**Comment by igorclark** [comment]
OK I'll see about giving that a try. Thanks @nicm!

**Comment by blueyed** [comment]
I am seeing a crash via `C-a C` (Customize options) in tmux 3.2a, which is fixed in add03dfb.
I am not using this feature in general, but might accidentally hit it with caps-lock being on.

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00005587a8c9103d in options_get_only (oo=oo@entry=0x0, name=name@entry=0x5587aa40c6a0 "@has_less") at options.c:220
220		found = RB_FIND(options_tree, &oo->tree, &o);
(gdb) bt full
#0  0x00005587a8c9103d in options_get_only (oo=oo@entry=0x0, name=name@entry=0x5587aa40c6a0 "@has_less") at options.c:220
        o = <optimized out>
        found = <optimized out>
#1  0x00005587a8c91131 in options_get (oo=oo@entry=0x0, name=0x5587aa40c6a0 "@has_less") at options.c:233
        o = <optimized out>
#2  0x00005587a8cc871b in window_customize_build_options (data=data@entry=0x5587aa43b310, title=title@entry=0x5587a8cebda7 "Server Options", tag=tag@entry=13835058055282163715, 
    scope0=scope0@entry=WINDOW_CUSTOMIZE_SERVER, oo0=0x5587aa3d4010, scope1=scope1@entry=WINDOW_CUSTOMIZE_NONE, oo1=0x0, scope2=WINDOW_CUSTOMIZE_NONE, oo2=0x0, ft=0x5587aa4a3640, filter=0x0, fs=0x7ffe14990d30)
    at window-customize.c:405
        top = 0x5587aa42c850
        o = <optimized out>
        loop = <optimized out>
        list = 0x5587aa438df0
        name = <optimized out>
        size = 2
        i = 0
        scope = <optimized out>
#3  0x00005587a8cc895c in window_customize_build (modedata=0x5587aa43b310, sort_crit=<optimized out>, tag=<optimized out>, filter=0x0) at window-customize.c:551
        data = 0x5587aa43b310
        fs = {flags = 0, current = 0x5587a8c8de9a <mode_tree_start+378>, s = 0x5587aa489b50, wl = 0x5587aa422890, w = 0x5587aa486480, wp = 0x5587aa486700, idx = -1}
        ft = 0x5587aa4a3640
        i = <optimized out>
        kt = <optimized out>
#4  0x00005587a8c8dfe7 in mode_tree_build (mtd=0x5587aa434110) at mode-tree.c:490
        s = 0x5587aa4341c0
        tag = 18446744073709551615
#5  0x00005587a8cc6fa5 in window_customize_init (wme=0x5587aa49ff90, fs=0x5587aa42df08, args=0x5587aa3f26d0) at window-customize.c:897
        wp = 0x5587aa486700
        data = 0x5587aa43b310
        s = 0x5587aa4341c0
#6  0x00005587a8cd0b84 in window_pane_set_mode (wp=wp@entry=0x5587aa486700, swp=swp@entry=0x0, mode=0x5587a8d15ae0 <window_customize_mode>, fs=fs@entry=0x5587aa42df08, args=args@entry=0x5587aa3f26d0)
    at window.c:1091
        wme = 0x5587aa49ff90
```

Would be great if this could make it into a bugfix release, since it is really nasty to lose like 20 windows.
